src/main/drivers/adc_stm32f4xx.c

```diff
@@ -125,17 +125,28 @@ void adcInit(const adcConfig_t *config)
 
     bool adcActive = false;
     for (int i = 0; i < ADC_CHANNEL_COUNT; i++) {
-        if (!adcVerifyPin(adcOperatingConfig[i].tag, device)) {
-            continue;
+        if (i < ADC_CHANNEL_COUNT_EXTERNAL) {
+            if (!adcVerifyPin(adcOperatingConfig[i].tag, device)) {
+                continue;
+            }
+
+            IOInit(IOGetByTag(adcOperatingConfig[i].tag), OWNER_ADC_BATT + i, 0);
+            IOConfigGPIO(IOGetByTag(adcOperatingConfig[i].tag), IO_CONFIG(GPIO_Mode_AN, 0, GPIO_OType_OD, GPIO_PuPd_NOPULL));
+            adcOperatingConfig[i].adcChannel = adcChannelByTag(adcOperatingConfig[i].tag);
+        } else {
+            if (i == ADC_TEMPERATURE) {
+                adcOperatingConfig[i].adcChannel = ADC_Channel_TempSensor;
+            } else if (i == ADC_VREFINT) {
+                adcOperatingConfig[i].adcChannel = ADC_Channel_Vrefint;
```

Comment:
Maybe `ADC_Channel_VRefInt`?

Reply:
`ADC_Channel_Vrefint` works.

Reply:
Ah, I didn't realise that this was a library variable and not one that you'd defined.